debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnection.java

```diff
@@ -71,7 +81,58 @@ else if ("true".equals(legacyDateTime)) {
         jdbcConfig = jdbcConfigBuilder.build();
         String driverClassName = jdbcConfig.getString(MySqlConnectorConfig.JDBC_DRIVER);
         this.jdbc = new JdbcConnection(jdbcConfig,
-                JdbcConnection.patternBasedFactory(MYSQL_CONNECTION_URL, driverClassName, getClass().getClassLoader()));
+                JdbcConnection.patternBasedFactory(MYSQL_CONNECTION_URL, driverClassName, getClass().getClassLoader()));*/
+    }
+
+    private static void mySqlConnectionSettings(Configuration.Builder builder) {
+        builder.with("useSSL", Boolean.toString(sslModeEnabled()));
+        if (DRIVER_MAJOR_VERSION_PRE8)
+            builder.with("zeroDateTimeBehavior", "CONVERT_TO_NULL");
+        else
+            builder.with("zeroDateTimeBehavior", "convertToNull");
+        final String legacyDateTime = jdbcConfig.getString(JDBC_PROPERTY_LEGACY_DATETIME);
```

Comment:
This is the bit where I had a question. Is these any ways we can pass the config objects to these lambdas? I know I can create a method and create a new builder here. But, that would mean not leveraging whatever's coming from upstream. Any examples of this? Also, couple of issues with the static non-static methods which also is a bit tricky.

Reply:
I don't fully understand what you are asking. WDYM by "upstream" here? In general, no instance methods should be called from the constructor, so you'd e.g. have to pass the config object to this method if you want to obtain values from it.

Reply:
I am sorry should have been more elaborative. By "upstream" I meant, the Operations FunctionInterface that has been defined. I tried to mimic the PostgresConnection class which creates the validateServerVersion by passing in a Statement object. Once the super constructor runs, then it is assigned to the initialOps object and executed. If I create a separate method and pass in the Config object to it, I will have to create all these objects. I hope I could elaborate.

Reply:
Ah, I'm beginning to understand. `Operations initialOperations` is meant for executing initial JDBC statements, though (e.g. for setting some session variables), not for initializing the connection's state (configuration etc.). I.e. you shouldn't try and set the config variables within `Operations`. What's the reasoning for not leaving that as is?